Sources/TuistLoader/Loaders/CachedManifestLoader.swift

```diff
@@ -101,6 +100,13 @@ public class CachedManifestLoader: ManifestLoading {
     // MARK: - Private
 
     private func load<T: Codable>(manifest: Manifest, at path: AbsolutePath, loader: () throws -> T) throws -> T {
+        if case .config = manifest, cacheDirectory == nil {
+            return try loader()
+        } else if cacheDirectory == nil {
+            let config = try TuistGraph.Config.from(manifest: try loadConfig(at: path), at: path)
+            cacheDirectory = CacheDirectoriesProviderFactory().cacheDirectories(config: config).manifestCacheDirectory
```

Comment:
factory used here should be also injected

Reply:
fixed